### ISSUE TITLE
[Web]: Enable Accessibility on Page Load

### ISSUE THREAD

**Issue by 838** [flutter/flutter#52104]
## Use case

To navigate through contents in a page with Screen Reader, the user most first enable Accessibility by pressing an invisible button on the page, and sometimes the button is not reachable by the screen reader.

## Proposal

Option to enable Accessibility on page load.

**Comment by yjbanov** [comment]
Can you please provide the operating system, browser, and the screen reader you tried it on? The accessibility button should always be reachable.

**Comment by 838** [comment]
Hi @yjbanov, 

Env: 
OS: macOS 10.15.3
Screen reader: VoiceOver
Browser: Chrome / Safari

I have tried both on a local project and on [gallery app](https://flutter.github.io/gallery/#/).

On safari is much more robust that the “Enable Accessibility button” is reachable/focused on page load/reload.
However on Chrome sometimes(ca. 1 out of 3 attempts) when I load/reload the page this can happen: 

1. Reload page by cmd+r
2. VoiceOver announces:
   <img height="100" alt="Activity" src="https://user-images.githubusercontent.com/3314171/76619514-5dcdf800-652b-11ea-80da-d2c67e3063df.png">
3. Pressing Control-Option-Shift-Down Arrow to enter the web area
4. VoiceOver announces: “Enable accessibility, group” instead of button, and nothing will happen if I press VO-right/left or VO-shift-DownArrow to enter the group
    <img height="100" alt="Displaying a" src="https://user-images.githubusercontent.com/3314171/76619518-60c8e880-652b-11ea-9587-de447f72c3b0.png">

5. The only way to enable accessibility is to reload the page(and hope that the enable button will be reachable) or reach the button by using “Item Chooser”

    ￼<img height="100" alt="Item Chooser" src="https://user-images.githubusercontent.com/3314171/76619519-61617f00-652b-11ea-86d2-d708785dc113.png">


I'm aware of that sometime ScreenReaders could act unexpected, therefore I'm proposing that if it could be possible in flutter, we could have an option to have Accessibility enabled by default, so the users don't need to enable the accessibility every time they visit the page.

**Comment by yjbanov** [comment]
Can you please provide the output of `flutter doctor --verbose`?

/cc @nturgut

**Comment by 838** [comment]
I have now tested on both beta and master channel and still same issue. 

You can reproduce the issue every time by following this steps on Chrome browser with VoiceOver enabled: 
1. Navigate to the page or reload the page if you are already there(after load VO will probably announce that you are on a web content)
2. press: ctrl+option+shift+up arrow to go out of web content 
3. press: ctrl+option+shift+down arrow to go in the web content again
4. press: ctrl+option+right arrow to go to next element 
5. VO will announce "Enable accessibility, Group" 

### Master channel 
```
[✓] Flutter (Channel master, v1.16.2-pre.25, on Mac OS X 10.15.3 19D76, locale en-DK)
    • Flutter version 1.16.2-pre.25 at /Users/b044560/Development/flutter
    • Framework revision ec8448d0a3 (4 hours ago), 2020-03-20 01:46:03 -0400
    • Engine revision fd60ddf751
    • Dart version 2.8.0 (build 2.8.0-dev.16.0 5b21a43c55)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/b044560/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /Users/b044560/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.9.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 44.0.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.43.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.8.1

[✓] Connected device (2 available)
    • Chrome     • chrome     • web-javascript • Google Chrome 80.0.3987.149
    • Web Server • web-server • web-javascript • Flutter Tools

• No issues found!
```

### Beta channel 
```
[✓] Flutter (Channel beta, v1.15.17, on Mac OS X 10.15.3 19D76, locale en-DK)
    • Flutter version 1.15.17 at /Users/b044560/Development/flutter
    • Framework revision 2294d75bfa (2 weeks ago), 2020-03-07 00:28:38 +0900
    • Engine revision 5aff311948
    • Dart version 2.8.0 (build 2.8.0-dev.12.0 9983424a3c)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/b044560/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /Users/b044560/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.9.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 44.0.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.43.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.8.1

[✓] Connected device (2 available)
    • Chrome     • chrome     • web-javascript • Google Chrome 80.0.3987.149
    • Web Server • web-server • web-javascript • Flutter Tools

• No issues found!
```

**Comment by nturgut** [comment]
Hi,

I tested the functionality on MacOS(10.14.6) Chrome (80.0.3987.132) using VoiceOver a few times, I didn't encounter issues.

After entering the content area, VoiceOver announces the "Enable accessibility button" as the first element. I am using VO+Space to press the button.

On the other hand I understand the user experience would be significantly better if semantics tree was always enabled. However enabled accessibly reduces the performance for Flutter for Web. We are currently not confident for always having semantics tree on.

Let me explore some options for building the app a11y enabled. Or maybe removing the accessibility button in a later stage in the desktop version which might make enabling more easier.

Thanks for filing the issue.

**Comment by adil-s** [comment]
I understand that enabling accessibility disables the performance. but could you add a check to enable accessibility when a screen reader is enabled? enabling accessibility via a prompt is not the most desireable 
 experience.

**Comment by nturgut** [comment]
Unfortunately we can't add that check. Due to privacy "whether screen reader on/off" is not accessible to browser via any API.

Under some cases (such as ChromeVox, Chromer extention) we can see extra css added to the page, however most of the time this screenreader information is private.
